bolt/include/bolt/Core/MCPlusBuilder.h

```diff
@@ -1888,6 +1888,19 @@ class MCPlusBuilder {
     llvm_unreachable("not implemented");
   }
 
+  /// Checks if the indirect call / jump is accepted by the landing pad at the
+  /// start of the target BasicBlock.
+  virtual bool isBTIVariantCoveringCall(MCInst &Call, MCInst &Pad) const {
+    llvm_unreachable("not implemented");
+    return false;
+  }
+
+  /// Adds a BTI landing pad to the start of the BB, that matches the indirect
+  /// call/jump inst.
+  virtual void addBTItoBBStart(BinaryBasicBlock &BB, MCInst &Call) const {
```

Comment:
Nit: should it be addBTIToBBStart?

If we're entertaining a rename, how about `insert` rather than add - since normally I think of putting something at the beginning of a list as 'insertion' rather than 'addition'. `insertBTI` would be a clear verb for a function name (and 'ToBBStart would be a bit redundant since presumably we won't expect to insert a BTI anywhere else?)

Reply:
ensureBTITarget?
makeBTITarget?

On the other hand `insertBTI` I think should be reasonably clear. If it begins with `if (alreadyBTITarget) return;` that shouldn't be too surprising to a developer given the purpose of BTI. At least I can't think of a good reason to need to double them up.